packages/runtime/src/next/lib/next.ts

```diff
@@ -0,0 +1,30 @@
+// Next.js 15 compatibility layer:
+//
+// `draftMode`, `cookies` and `params` functions became async in Next.js 15
+// (https://nextjs.org/blog/next-15#async-request-apis-breaking-change),
+// but we are supporting Next.js 13, 14 and 15 in the same codebase, and are
+// compiling against Next.js 14 as a common denominator. The wrapper
+// functions below force the async use of the aforementioned primitives in
+// the rest of the codebase even though they are typed as synchronous in
+// Next.js 14. This works because awaiting a non-promise value is a no-op in
+// JavaScript.
```

Comment:
Why is this compatibility layer necessary? Can't we just do `await cookies()` or `await draftMode()` whether we're on Next.js 14 or 15?

Reply:
@migueloller The comment explains it:

> The wrapper functions below force the async use of the aforementioned primitives in the rest of the codebase even though they are typed as synchronous in Next.js 14.

Reply:
Sorry, hahaha. I read it but it didn't make sense to me so I was compelled to ask. I think I was looking for a user-facing reason, not a maintainer-facing one.

So my updated understanding here is that this is to avoid us forgetting to `await` the new APIs, not as utilities we actually export? If so, could we accomplish the same with the ESLint rule for [no floating promises](https://typescript-eslint.io/rules/no-floating-promises/)?

I feel like having the utility helps if you know the utility exists, but a maintainer could mistakenly import from `next` and that could be missed in code review.

Reply:
Yep, I think I'm gonna go with the latest after all.